core/src/test/java/org/apache/calcite/test/SqlJsonFunctionsTest.java

```diff
@@ -533,6 +533,13 @@ public void testJsonKeys() {
         is("null"));
   }
 
+  @Test
+  public void testJsonRemove() {
+    assertJsonRemove(is("{\"b\":[2]}"),
+              SqlFunctions.dejsonize("{\"a\": 1, \"b\": [2]}"),
```

Comment:
These `asssertXxx` methods are messed up. The point of the `assertThat` style is that you can read it like a sentence: "assert that <value> is <predicate>". So, the matcher has to be the last argument. Yes, even if that means you can't use varargs.

Reply:
And maybe we can add something like the `SqlTester#setFor()` method (see below) for both `SqlFunctuionsTest` and `SqlJsonFunctionsTest` for generating readable messages, so that we can remove these `assertXXX` methods.

https://github.com/apache/calcite/blob/72e952d1a79ee2d7ba05de88cbc2ac11f65cd879/core/src/test/java/org/apache/calcite/sql/test/SqlTester.java#L298-L300

Reply:
well,there is a lot going on in the `JsonFunctions` section, I feel the need to extract `SqlJsonFunctionsTest` as `JsonFunction's` test.